File: libs/utils/lib-core/src/main/java/com/yyxnb/what/core/log/LogUtils.java
package com.yyxnb.what.core.log;

import java.util.List;
import java.util.Map;

public final class LogUtils {

    private static final IPrinter LOGGER_PRINTER = new LoggerPrinter();

    private LogUtils() {

    }

    public static LogConfig init() {
        return LOGGER_PRINTER.init();
    }

    public static String getFormatLog() {
        return LOGGER_PRINTER.getFormatLog();
    }

    public static void d(String message, Object... args) {
        LOGGER_PRINTER.d(message, args);
    }

    public static void e(String message, Object... args) {
        LOGGER_PRINTER.e(null, message, args);
    }

    public static void e(Throwable throwable, String message, Object... args) {
        LOGGER_PRINTER.e(throwable, message, args);
    }

    public static void i(String message, Object... args) {
        LOGGER_PRINTER.i(message, args);
    }

    public static void v(String message, Object... args) {
        LOGGER_PRINTER.v(message, args);
    }

    public static void w(String message, Object... args) {
        LOGGER_PRINTER.w(message, args);
    }

    public static void wtf(String message, Object... args) {
        LOGGER_PRINTER.wtf(message, args);
    }

    public static void json(String json) {
        LOGGER_PRINTER.json(json);
    }

    public static void xml(String xml) {
        LOGGER_PRINTER.xml(xml);
    }

    public static void map(Map map) {
        LOGGER_PRINTER.map(map);
    }

    public static void list(List list) {
        LOGGER_PRINTER.list(list);
    }

}

File: lib_common/src/main/java/com/yyxnb/common/utils/log/LogUtils.kt
package com.yyxnb.common.utils.log

object LogUtils {

    private val LOGGER_PRINTER: IPrinter = LoggerPrinter

    fun init(): LogConfig {
        return LOGGER_PRINTER.init()
    }

    val formatLog: String
        get() = LOGGER_PRINTER.formatLog

    @JvmStatic
    fun d(message: String?, vararg args: Any) {
        LOGGER_PRINTER.d(message, *args)
    }

    @JvmStatic
    fun e(message: String?, vararg args: Any) {
        LOGGER_PRINTER.e(null, message, *args)
    }

    fun e(throwable: Throwable?, message: String?, vararg args: Any) {
        LOGGER_PRINTER.e(throwable, message, *args)
    }

    fun i(message: String?, vararg args: Any) {
        LOGGER_PRINTER.i(message, *args)
    }

    fun v(message: String?, vararg args: Any) {
        LOGGER_PRINTER.v(message, *args)
    }

    @JvmStatic
    fun w(message: String?, vararg args: Any) {
        LOGGER_PRINTER.w(message, *args)
    }

    fun wtf(message: String?, vararg args: Any) {
        LOGGER_PRINTER.wtf(message, *args)
    }

    fun json(json: String) {
        LOGGER_PRINTER.json(json)
    }

    fun xml(xml: String) {
        LOGGER_PRINTER.xml(xml)
    }

    fun map(map: Map<*, *>?) {
        LOGGER_PRINTER.map(map)
    }

    @JvmStatic
    fun list(list: List<*>?) {
        LOGGER_PRINTER.list(list)
    }
}